proxy/src/transport/tls/connection.rs

```diff
@@ -0,0 +1,78 @@
+use std::io;
+use std::net::SocketAddr;
+
+use bytes::Buf;
+use futures::Future;
+use tokio::prelude::*;
+use tokio::net::TcpStream;
+
+use transport::{AddrInfo, Io};
+
+use super::{
+    rustls::ServerSession,
+    tokio_rustls::{ServerConfigExt, TlsStream},
+
+    ServerConfig,
+};
+
+// In theory we could replace `TcpStream` with `Io`. However, it is likely that
+// in the future we'll need to do things specific to `TcpStream`, so optimize
+// for that unless/until there is some benefit to doing otherwise.
+#[derive(Debug)]
+pub struct Connection(TlsStream<TcpStream, ServerSession>);
+
+impl Connection {
+    pub fn accept(socket: TcpStream, ServerConfig(config): ServerConfig)
+                  -> impl Future<Item = Connection, Error = io::Error>
+    {
+        config.accept_async(socket).map(Connection)
```

Comment:
I think we might eventually want to add a `::logging::context_future` to this future with a `tls=true` field or something similar.

Reply:
I'm don't think that the accept future this is the right place to do this.

Reply:
> I think we might eventually want to add a ::logging::context_future to this future with a tls=true field or something similar.

Definitely we need to figure out how to get the right context information into the logging messages produced by tokio-rustls and Rustls, and by the TLS submodule during the handshake. I could use some help with that as I'm not familiar with the latest logging stuff in Conduit.